server/src/test/java/io/spine/examples/shareaware/server/investment/SharesPurchaseTest.java

```diff
@@ -0,0 +1,305 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.server.investment;
+
+import io.spine.core.UserId;
+import io.spine.examples.shareaware.InvestmentId;
+import io.spine.examples.shareaware.ShareId;
+import io.spine.examples.shareaware.WalletId;
+import io.spine.examples.shareaware.investment.Investment;
+import io.spine.examples.shareaware.investment.SharesPurchase;
+import io.spine.examples.shareaware.investment.command.AddShares;
+import io.spine.examples.shareaware.investment.command.PurchaseShares;
+import io.spine.examples.shareaware.investment.event.SharesPurchaseFailed;
+import io.spine.examples.shareaware.investment.event.SharesPurchased;
+import io.spine.examples.shareaware.market.command.ObtainShares;
+import io.spine.examples.shareaware.server.investment.given.PurchaseTestContext;
+import io.spine.examples.shareaware.server.investment.given.RejectingMarket;
+import io.spine.examples.shareaware.wallet.Wallet;
+import io.spine.examples.shareaware.wallet.command.CancelMoneyReservation;
+import io.spine.examples.shareaware.wallet.command.DebitReservedMoney;
+import io.spine.examples.shareaware.wallet.command.ReserveMoney;
+import io.spine.examples.shareaware.wallet.event.MoneyReservationCanceled;
+import io.spine.examples.shareaware.wallet.event.MoneyReserved;
+import io.spine.examples.shareaware.wallet.event.ReservedMoneyDebited;
+import io.spine.examples.shareaware.wallet.rejection.Rejections.InsufficientFunds;
+import io.spine.server.BoundedContextBuilder;
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import io.spine.testing.server.model.ModelTests;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.examples.shareaware.server.investment.given.InvestmentTestEnv.*;
+import static io.spine.examples.shareaware.server.given.WalletTestEnv.setUpReplenishedWallet;
+import static io.spine.examples.shareaware.server.given.WalletTestEnv.setUpWallet;
+
+@DisplayName("`SharesPurchase` should")
+public final class SharesPurchaseTest extends ContextAwareTest {
+
+    @BeforeAll
+    static void beforeAll() {
+        ModelTests.dropAllModels();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        ModelTests.dropAllModels();
+    }
+
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return PurchaseTestContext.newBuilder();
+    }
+
+    @Nested
+    @DisplayName("reduce the wallet balance")
+    class WithdrawMoneyFromBalance {
+
+        @Test
+        @DisplayName("for the purchase price amount")
+        void event() {
```

Comment:
There is no event check in this method.
```suggestion
        void walletState() {
```